### PR TITLE
TOOL-22158 Add new delete-old-branches action

### PR DESCRIPTION
**Pull request by ghost**
<details open>
<summary><h2> Problem </h2></summary>

We'd like a way to delete old unused branches

</details>

<details open>
<summary><h2> Solution </h2></summary>

Create a GitHub action that developers can add to their repositories
to perform cleanup for them

</details>

<details>
<summary><h2> Testing Done </h2></summary>

I've run the code locally using `act` against this repository:
```
| DRY RUN: Deleting dlpx/pr/prakashsurya/41d182b0-3fe9-4b4d-839b-bfcd389d7f21
| dlpx/pr/prakashsurya/4da81eaa-b0e9-4fc2-9644-bc10b6fa03b4 has been updated recently. Skipping
| dlpx/pr/prakashsurya/bdacf2d7-f702-488e-a382-d0cce5e0a95d has an open pull request. Skipping
| master is an excluded branch. Skipping
```
</details>

### CODE REVIEW

File: .github/workflows/delete-old-branches.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Cleanup old branches
+on:
+  push:
+    branches:
+      - master
```
Comment (robinrye) [inline .github/workflows/delete-old-branches.yaml:5]: does this need to be dynamic? we use main a lot these days right?

Comment (robinrye) [reply .github/workflows/delete-old-branches.yaml:5]: oh this is in the actions repo. makes sense.

**Review by robinrye — APPROVED** [review]

File: .github/workflows/delete-old-branches.yaml (hunk 1 above)
Comment (ghost) [reply .github/workflows/delete-old-branches.yaml:5]: Yeah exactly this file is it running in this repository, just one instance of the action. I wanted to add it here for testing and as an example

File: delete-old-branches/delete_old_branches.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        )
+
+    return response
+
+#
+# Branch checks
+#
+def _excluded_branch(branch: str) -> bool:
+    """
+    Return True if the branch is considered a default branch
```
Comment (grodr) [inline delete-old-branches/delete_old_branches.py]: *nit* this is checking if the branch is in excluded_branches, right, not a default branch a the repo? Should we update the description and variable name to reflect that?

**Review by grodr — APPROVED** [review]

File: delete-old-branches/delete_old_branches.py (hunk 2 above)
Comment (ghost) [reply delete-old-branches/delete_old_branches.py]: Good call TY

**Review by jorge-morse — APPROVED** [review]

**Pull request merged**